starcheck/plot.py

```diff
@@ -0,0 +1,187 @@
+    fig = plt.figure(figsize=figsize)
+    ax = fig.add_subplot(1, 1, 1)
+    face = backcolor
+
+    # plot the box and set the labels
+    plt.xlim(2900, -2900)
+    plt.ylim(-2900, 2900)
+    b1hw = 2560
+    box1 = plt.Rectangle((b1hw, -b1hw), -2 * b1hw, 2 * b1hw,
+                         fill=False)
+    ax.add_patch(box1)
+    b2hw = 2600
+    box2 = plt.Rectangle((b2hw, -b2hw), -2 * b2hw, 2 * b2hw,
+                         fill=False)
+    ax.add_patch(box2)
+
+    ax.scatter([-2700, -2700, -2700, -2700, -2700],
+               [2400, 2100, 1800, 1500, 1200],
+               c='orange', edgecolors='orange',
+               s=symsize(np.array([10.0, 9.0, 8.0, 7.0, 6.0])))
+
+    [l.set_rotation(90) for l in ax.get_yticklabels()]
+    ax.grid()
+    ax.set_ylabel("Zag (arcsec)")
+    ax.set_xlabel("Yag (arcsec)")
+
+    # plot starcheck catalog
+    if catalog is not None:
+        cat = Table(catalog)
+        gui = cat[cat['type'] == 'GUI']
+        acq = cat[cat['type'] == 'ACQ']
+        bot = cat[cat['type'] == 'BOT']
+        fid = cat[cat['type'] == 'FID']
+        for row in cat:
+            ax.annotate("%s" % row['idx'],
+                        xy=(row['yang'] - 120, row['zang'] + 60))
+        ax.scatter(gui['yang'], gui['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   s=100)
+        ax.scatter(bot['yang'], bot['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   s=100)
+        for acq_star in acq:
+            acq_box = plt.Rectangle(
+                (acq_star['yang'] - acq_star['halfw'],
+                 acq_star['zang'] - acq_star['halfw']),
+                width=acq_star['halfw'] * 2,
+                height=acq_star['halfw'] * 2,
+                color='blue',
+                fill=False)
+            ax.add_patch(acq_box)
+        for acq_star in bot:
+            acq_box = plt.Rectangle(
+                (acq_star['yang'] - acq_star['halfw'],
+                 acq_star['zang'] - acq_star['halfw']),
+                width=acq_star['halfw'] * 2,
+                height=acq_star['halfw'] * 2,
+                color='blue',
+                fill=False)
+            ax.add_patch(acq_box)
+        ax.scatter(fid['yang'], fid['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   marker='o',
+                   s=200)
+        ax.scatter(fid['yang'], fid['zang'],
+                   facecolors=face,
+                   edgecolors='green',
+                   marker='+',
+                   linewidth=3,
+                   s=200)
+
+    # plot field if present
+    if field is not None:
+        faint_plot_mag = 11.0
+        bright_field = field[field['MAG_ACA'] < faint_plot_mag]
+        yags = []
+        zags = []
+        for star in bright_field:
+            yag, zag = radec2yagzag(star['RA_PMCORR'],
```

Comment:
Always search for ways to use functional forms and tables.
```
# Add star Y angle and Z angle in arcsec to the field_stars table
yagzags = (radec2yagzag(star['RA_PMCORR'], star['DEC_PMCORR'], attitude) 
           for star in field_stars)
yagzags = Table(rows=[(y * 3600, z * 3600) for y, z in yagzags], names=['yag', 'zag'])
field_stars = table.hstack([field_stars, yagzags])
```
So 4 lines with a comment that clearly states what is being done.